consensus/src/chained_bft/consensus_types/block.rs

```diff
@@ -106,6 +108,30 @@ pub struct Block<T> {
     block_source: BlockSource,
 }
 
+/// ExecutedBlocks are managed in a speculative tree, the committed blocks form a chain. Besides
+/// block data, each executed block also has other derived meta data which could be regenerated from
+/// blocks.
+#[derive(Clone, Debug)]
+pub struct ExecutedBlock<T> {
+    /// Block data that cannot be regenerated.
+    block: Arc<Block<T>>,
+    /// The set of children for cascading pruning. Note: a block may have multiple children.
+    children: HashSet<HashValue>,
+    /// The state compute results is calculated for all the pending blocks prior to insertion to
+    /// the tree (the initial root node might not have it, because it's been already
+    /// committed). The execution results are not persisted: they're recalculated again for the
+    /// pending blocks upon restart.
+    compute_result: Arc<StateComputeResult>,
+}
+
+impl<T> Deref for ExecutedBlock<T> {
```

Comment:
this is not encouraged to use other than a pointer. where do you need this?

Reply:
Yep, maybe I should remove it.